Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    int numAliveNeighs = GameOfLife.calcAliveNeighs(board, i, j, rows, cols);
+                    if (board[i, j])
+                    {
+                        switch(numAliveNeighs)
+                        {
+                            case 0: case 1: case 4:
+                                newBoard[i, j] = false;
+                                break;
+                            case 2: case 3:
+                                newBoard[i, j] = true;
+                                break;
+                            default:
+                                Console.WriteLine("Shouldn't get here.");
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        if (numAliveNeighs == 3)
+                            newBoard[i, j] = true;
+                        else
+                            newBoard[i, j] = false;
+                    }
+                }
+                
+            }
+            // print output matrix
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    Console.Write(string.Format("{0} ", newBoard[i, j]));
+                }
+                Console.Write(Environment.NewLine + Environment.NewLine);
+            }
         }
+            
 
         public static int Assignment9()
         {
-            /*
-             * TODO: Write code here
-             */
+            const double densityThresh = 0.99;
+            double density = 0;
+            int currNum = 0;
+            int jumpingCount = 0;
+            while (density < densityThresh)
+            {
+                currNum += 1;
+                if (BouncingNumbers.IsJumpingNumber(currNum))
+                    jumpingCount++;
+                density = (double)jumpingCount / currNum;
+            }
+            return currNum;
+                
         }
         #endregion
 
         #region Bonus solutions
         public static int Bonus1()
         {
-            /*
-             * TODO: Write code here
-             */
+            int totalCount = 0;
+            string[] oneToNine = new string[]{"one", "two", "three", "four", "five", "six", "seven",
+                                                "eight", "nine"};
+            string[] tenToNineteen = new string[]{"ten", "eleven", "twelve", "thirteen", "fourteen",
+                                                    "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
+            string[] twentyToNinety = new string[]{"twenty", "thirty", "forty", "fifty", "sixty", "seventy",
+                                                    "eighty", "ninety"};
+            int oneToNineCount = 0;
+            foreach (string num in oneToNine)
+                oneToNineCount += num.Length;
```

Comment:
Why not add straight to `totalCount`?

Reply:
To add that number 10 times for the the 'one' 'two'...'nine' in 21,..,29 etc.